pipeline/transform.py

```diff
@@ -0,0 +1,17 @@
+"""Transforms the plant data to make it ready for loading"""
+import pandas as pd
+
+
+def clean_plant_data(plant_df: pd.DataFrame) -> pd.DataFrame:
+    """Setting the types for dataframe columns and removing NaN values"""
+
+    plant_df['recording_at'] = pd.to_datetime(plant_df['recording_at'])
+    plant_df['last_watered'] = pd.to_datetime(plant_df['last_watered'])
+    plant_df['soil_moisture'] = pd.to_numeric(plant_df['soil_moisture'], errors='COERCE')
+    plant_df['temperature'] = pd.to_numeric(plant_df['temperature'])
+    plant_df['plant_id'] = pd.to_numeric(plant_df['plant_id'])
+    plant_df = plant_df[(plant_df['soil_moisture'] >= 0) &
+                        (plant_df['soil_moisture'] <= 100)]
+
```

Comment:
Could you add this line to clean the plant names because some have trailing white space or "" in the names
plant_df['plant_name'] = plant_df['plant_name'].str.strip(", ")